spark/src/test/scala/org/apache/spark/sql/delta/CustomCatalogSuite.scala

```diff
@@ -366,3 +382,56 @@ class DummyCatalog extends TableCatalog {
     )
   }
 }
+
+// A dummy catalog that adds additional table storage properties after the table is loaded.
+// It's only used inside `DummySessionCatalog`.
+class DummySessionCatalogInner extends DelegatingCatalogExtension {
+  override def loadTable(ident: Identifier): Table = {
+    val t = super.loadTable(ident).asInstanceOf[V1Table]
+    V1Table(t.v1Table.copy(
+      storage = t.v1Table.storage.copy(
+        properties = t.v1Table.storage.properties ++ Map("fs.myKey" -> "val")
```

Comment:
nit: Does this work?
```suggestion
        properties = t.v1Table.storage.properties.updated("fs.myKey" -> "val")
```

Reply:
not sure, we are still on scala 2.12 and I feel it's safer to use common APIs.

Reply:
[Map.updated](https://www.scala-lang.org/api/2.12.7/scala/collection/Map.html#updated(key:K,value:V):scala.collection.Map[K,V]) is a scala-2.12 API..